File: informatika_ismeretek/kozep/2021_okt/java/Elethossz.java
public class Elethossz {
    private int Tol;
    private int Ig;

    public int elethosszEvekben() {
        return Tol == -1 || Ig == -1 ? -1 : Ig - Tol;
    }

    public boolean ismertElethossz() {
        return elethosszEvekben() != -1;
    }

    public Elethossz(String szuletesHalalozas) {
        String[] m = szuletesHalalozas.split("-");
        try {
            Tol = Integer.parseInt(m[0]);
        } catch (Exception e) {
            Tol = -1;
        }
        try {
            Ig = Integer.parseInt(m[1]);
        } catch (Exception e) {
            Ig = -1;
        }
    }
}

File: informatika_ismeretek/kozep/2021_okt/c#/Elethossz.cs
class Elethossz
{
    private int Tol { get; set; }
    private int Ig { get; set; }
    public int  ElethosszEvekben => Tol == -1 || Ig == -1 ? -1 : Ig - Tol;

    public bool IsmertAzElethossz => ElethosszEvekben != -1;

    public Elethossz(string SzuletesHalalozas)
    {
        string[] m = SzuletesHalalozas.Split('-');
        try
        {
            Tol = int.Parse(m[0]);
        }
        catch (Exception)
        {
            Tol = -1;
        }
        try
        {
            Ig = int.Parse(m[1]);
        }
        catch (Exception)
        {
            Ig = -1;
        }
    }
}